src/formatter/ExpressionFormatter.ts

```diff
@@ -45,12 +45,21 @@ interface ExpressionFormatterParams {
 export interface DialectFormatOptions {
   // List of operators that should always be formatted without surrounding spaces
   alwaysDenseOperators?: string[];
+  // List of clauses that should be formatted on a single line
+  onelineClauses: string[];
+}
+
+// Contains the same data as DialectFormatOptions,
+// but optimized for faster and more conventient lookup.
+export interface ProcessedDialectFormatOptions {
+  alwaysDenseOperators: string[];
+  onelineClauses: Record<string, boolean>;
```

Comment:
can be
```suggestion
  onelineClauses: Record<string, true>;
```

Reply:
Tried that, but in this case TypeScript thinks that `onlineClauses[anything]` always evaluates to true.